### PR TITLE
bpo-26053: Fix args echoed by pdb run command

### PR DESCRIPTION
**Pull request by iritkatriel**
https://bugs.python.org/issue26053

### CODE REVIEW

**Comment by iritkatriel** [comment]
@birkenfeld are you able to review this?

**Comment by birkenfeld** [comment]
Sorry, not at the moment.

File: Misc/NEWS.d/next/Library/2020-09-01-10-12-13.bpo-26053.hXikw_.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed args echoed by the pdb interactive run command.
```
Comment (ZackerySpytz) [inline Misc/NEWS.d/next/Library/2020-09-01-10-12-13.bpo-26053.hXikw_.rst]:
```rst
:mod:`pdb`
```
Also, I think this news entry needs more details.

Comment (iritkatriel) [reply Misc/NEWS.d/next/Library/2020-09-01-10-12-13.bpo-26053.hXikw_.rst]: Thank you, I've edited it. Is this better?

File: Lib/test/test_pdb.py [hunk 2]
```diff
@@ -1443,6 +1443,19 @@ def test_issue16180(self):
...
+        commands = """
+            continue
+            run a b c
+            run d e f
+            quit
+        """
+        stdout, stderr = self.run_pdb_script(script, commands)
+        output = ';'.join([x.strip() for x in stdout.splitlines()])
+        self.assertIn("Restarting main.py with arguments:;a b c", output)
+        self.assertIn("Restarting main.py with arguments:;d e f", output)
```
Comment (taleinat) [inline Lib/test/test_pdb.py]: I find the use of ';' here rather confusing, why not '\n'?

Comment (iritkatriel) [reply Lib/test/test_pdb.py]: You're right, \n would be consistent with test_errors_in_command

**Review by taleinat — APPROVED** [review]
LGTM!

Since I've not worked on pdb before, I'm still considering whether this should be backported.

**Comment by gvanrossum** [comment]
This looks like a real bug fix and I am fine with backporting to 3.9 and 3.8.

**Comment by iritkatriel** [comment]
Shall we merge this?

**Comment by gvanrossum** [comment]
Does it need a backport?

**Comment by iritkatriel** [comment]
The bug is in 3.9 and 3.9 as well.

**Comment by iritkatriel** [comment]
I'm checking the build issue.

**Comment by vstinner** [comment]
> I'm checking the build issue.

Also, why test_gdb didn't catch it in our fleet of CI jobs? gdb is not installed on the Ubuntu job for example?

**Comment by iritkatriel** [comment]
Following up here: https://github.com/python/cpython/pull/25139

@bedevere-bot, you make a good point but you're coming on a bit strong.

**Pull request merged**